src/eventbridge/bus.ts

```diff
@@ -0,0 +1,114 @@
+import { aws_events } from "aws-cdk-lib";
+import { Construct } from "constructs";
+import { EventBusRule, EventPredicateFunction } from "./rule";
+import { EventBusRuleInput } from "./types";
+
+/**
+ * A Functionless wrapper for a AWS CDK {@link aws_events.EventBus}.
+ *
+ * Filtering events and sending them to Lambda.
+ *
+ * ```ts
+ * interface Payload {
+ *    value: string;
+ * }
+ *
+ * // An event with the payload
+ * type myEvent = EventBusRuleInput<Payload>;
+ *
+ * const myAwsFunction = new aws_lambda.Function(this, 'myFunction', { ... });
+ * // A function that expects the payload.
+ * const myLambdaFunction = new Function<Payload, void>();
+ *
+ * // Some AWS Event Bus, may be new, may be default or imported.
+ * const awsBus = new aws_events.EventBus(this, "mybus");
+ * new EventBus<myEvent>(awsBus)
+ *    // when the payload is equal to some value
+ *    .when(this, 'rule1', event => event.detail.value === "some value")
+ *    // grab the payload
+ *    .map(event => event.detail)
+ *    // send to the function
+ *    .pipe(myLambdaFunction);
+ * ```
+ *
+ * Forwarding to another Event Bus based on some predicate:
+ *
+ * ```ts
+ * const anotherEventBus = aws_event.EventBus.fromEventBusArn(...);
+ *
+ * new EventBus<myEvent>(awsBus)
+ *    // when the payload is equal to some value
+ *    .when(this, 'rule2', event => event.detail.value === "some value")
+ *    // send verbatim to the other event bus
+ *    .pipe(anotherEventBus);
+ * ```
+ */
+export class EventBus<E extends EventBusRuleInput> {
+  /**
+   * This static property identifies this class as an EventBus to the TypeScript plugin.
+   */
+  public static readonly FunctionlessType = "EventBus";
+
+  constructor(readonly bus: aws_events.EventBus) {}
+
+  /**
+   * Filter events using as a EventBus Rule using a Functionless predicte function.
+   *
+   * Equals
+   *
+   * ```ts
+   * when(this, 'rule', (event) => event.source === "lambda")
```

Comment:
Explain the `this` and `'rule'`.

Also, use `"` instead of `'` to align with our ESLint configuration.